extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/JPAConfig.java

```diff
@@ -74,7 +75,7 @@ boolean isJtaEnabled() {
      *
      * @param event ignored
      */
-    void destroy(@BeforeDestroyed(ApplicationScoped.class) Object event) {
+    void destroy(@Observes @BeforeDestroyed(ApplicationScoped.class) Object event) {
```

Comment:
So this wasn't being called?

Reply:
It wasn't. That's what was missing to have `Integrator#disintegrate` being called :wink:

Reply:
ic. Maybe you have time to add an integration test for `#disintegrate` as well?